src/html5_parser/element_class.rs

```diff
@@ -0,0 +1,157 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ElementClass {
+    /// a map of classes applied to an HTML element.
+    /// key = name, value = is_active
+    /// the is_active is used to toggle a class (JavaScript API)
+    class_map: HashMap<String, bool>,
+}
+
+impl Default for ElementClass {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ElementClass {
+    /// Initialise a new (empty) ElementClass
+    pub fn new() -> Self {
+        ElementClass {
+            class_map: HashMap::new(),
+        }
+    }
+
+    /// Initialize a class from a class string
+    /// with space-delimited class names
+    pub fn from_string(class_string: &str) -> Self {
+        let mut class_map_local = HashMap::new();
+        let classes = class_string.split_whitespace();
+        for class_name in classes {
+            class_map_local.insert(class_name.to_owned(), true);
+        }
+
+        ElementClass {
+            class_map: class_map_local,
+        }
+    }
+
+    /// Count the number of classes (active or inactive)
+    /// assigned to an element
+    pub fn count(&self) -> usize {
```

Comment:
Wouldn't it be better to stick with the Rust style here and use `len` as the method name? Especially because there is an `is_empty` method, which gets suggested by clippy when doing any comparison to zero, e.g. `list.len() == 0`.

Reply:
yes, good suggestion. i'm still not entirely familiar with all of Rust's conventions yet. Will fix now

Reply:
updated in [98e6130](https://github.com/gosub-browser/gosub-engine/pull/92/commits/98e61304d3836a0769df4317afe5db9006af37e3)